src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -589,5 +602,30 @@ public void onPatchGamePrefsForMultipleInstances() {
       notificationService.addImmediateErrorNotification(e, "settings.fa.patchGamePrefsFailed");
     }
   }
+
+  public void onAddMirrorURL() {
+    String text = mirrorURITextField.getText();
+    if (text.isBlank()) {
+      return;
+    }
+    if (!text.endsWith("/")) {
+      text = text + "/";
+    }
+
+    URI uri = null;
+    try {
+      uri = new URL(text).toURI();
+    } catch (URISyntaxException | MalformedURLException e) {
+      log.debug("Failed to add invalid URL: {}", text, e);
+      return;
```

Comment:
Probably want a user notification here

Reply:
It would be nice if there were a way to flash an error message but I think a notification would be a bit overkill. You can tell that there is an error by the fact that the text field is not cleared but it would also be nice to show a small tooltip like message. Not sure if there’s a way to do that in JavaFX.

Reply:
I would just use the addimmeadiateerrornotification

Reply:
Err actually probably just addimmeadiatewarnnotification